### PR TITLE
Import export porting

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
This PR ports Import and Export plugins to main branch

Note: we should remove the ShapeFile plugin
TODO: 
 - [ ] Review text strings with @tdipisa

### CODE REVIEW

File: web/client/actions/mapimport.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2016, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const SET_LAYERS = 'IMPORT::SET_LAYERS';
```
Comment (mbarto) [inline web/client/actions/mapimport.js]: Why double colon?

**Comment by mbarto** [comment]
Aren't we replicating a lot of code / functionality that is already in the Shapefile plugin without removing / aliasing it?

**Comment by offtherailz** [comment]
@mbarto, yes I did the PR so we can discuss if remove or alias shapefile.

File: web/client/components/import/dragZone/enhancers/processFiles.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const Rx = require('rxjs');
+const { compose, mapPropsStream, createEventHandler} = require('recompose');
+const FileUtils = require('../../../../utils/FileUtils');
+const LayersUtils = require('../../../../utils/LayersUtils');
+
+const JSZip = require('jszip');
```
Comment (mbarto) [inline web/client/components/import/dragZone/enhancers/processFiles.jsx]: All this code is already in the Shapefile plugin, isn't it?

File: web/client/components/import/style/StylePanel.jsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
+const PropTypes = require('prop-types');
```
Comment (mbarto) [inline web/client/components/import/style/StylePanel.jsx]: Is this completely different than the styling panel used by the Shapefile plugin?

File: web/client/actions/mapimport.js (hunk 1 above)
Comment (offtherailz) [reply web/client/actions/mapimport.js]: Good catch, I will fix it together with shapefile stuff

File: web/client/components/import/style/StylePanel.jsx (hunk 3 above)
Comment (offtherailz) [reply web/client/components/import/style/StylePanel.jsx]: No, it is the same, but more isolated from import stuff, so we may be able to improve and reuse it in a easier way.

**Comment by offtherailz** [comment]
@tdipisa and me discussed about the opportunity to maintain the shapefile as a wrapper. 
We have decided that it is better to delete the shapefile plugin and add to the existing configuration Import and Export map plugin. 
We have to schedule time to 
 - remove ShapeFile plugin 
 - replace existing configurations with the new one
 - Document migration in migration guide

**Comment by MV88** [comment]
As a side note, the supported GeoJSON format does not support the type **Feature** but instead requires an object containing an array of Feature under the features props, i.e a "FeatureCollection" something like:
`{features: [{GeoJSON feature}, ...]}`
and by the way it does not check the mandatory presence of any **type, properties, and geometry** properties and it does check for example the Right hand rule
IMO: we should add/use a GeoJSON lint functionality when importing a GeoJSON Object

**Pull request closed without merging**